### ISSUE TITLE
Thoughts? Slippage = 0.0 -> 1.0 and no higher. Need custom Float Class?

### ISSUE THREAD

**Issue by bobsawey** [tinymanorg/tinyman-py-sdk#42]
Hello everyone, 

When we are submitting quotes with a slippage higher than 1.0 there is no error thrown from what I can tell.

But slippage requirement for submitting a transaction is 1.0...0 as the max float value.

Imho an error for slippage invalid amount should be thrown during quote operations and not at the point we are preparing a transaction.

BUT there is a bit of a hairy issue. From what I can tell after glancing over the source, slippage is defined in such a way where DRY deserves recognition (I rarely say this) -- but to avoid completely refactoring the pools code, the only way I can think of to do this with the least amount of intrusiveness is to write a custom float class and set slippage to the custom float in the SwapQuote dataclass definition.

Basically, I'm of the school of thought don't trust the programmer to assert and instead throw informative errors. I spent about an hour this morning circling wondering why my swaps were suddenly no longer working. The slippage variable in my source was recently updated to allow values up to 49.000000. After testing with slippage at around 3.32, the transaction group prep threw algosdk.error.WrongAmountType but the trace is a bit confusing as it pointed toward swap.py and from what I can tell the issue actually starts at pools.

@fergalwalsh what do you think? If this seems sane, I am in the code enough lately I'm fine with attempting to drum it out and submit a pull request.

**Comment by bobsawey** [comment]
Ah. I just realized dataclass fields may provide a simple path to resolution. new to dataclasses. Forked and reading up.
